modin/core/dataframe/pandas/partitioning/partition_manager.py

```diff
@@ -1431,6 +1445,11 @@ def rebalance_partitions(cls, partitions):
                 new_partitions.append(
                     cls.column_partitions(partitions[start : stop + 1], full_axis=False)
                 )
+                length = sum(part.length() for part in partitions[start : stop + 1, 0])
+                for col in new_partitions[-1]:
+                    col._length_cache = length
```

Comment:
This seems like it's repeating code like this: https://github.com/modin-project/modin/blob/c67324b04b01e7d1f27a7c91565c488537891de1/modin/core/execution/ray/implementations/pandas_on_ray/partitioning/virtual_partition.py#L376-L378

Do we need to have both?

Reply:
Good catch!